### PR TITLE
Use ember preview types

### PR DESCRIPTION
**Pull request by gitKrystan**
Depends on https://github.com/emberjs/ember-test-helpers/pull/1271
Should unblock fixing https://github.com/emberjs/ember.js/issues/20254

### CODE REVIEW

File: addon-test-support/@ember/test-helpers/-internal/build-registry.ts [hunk 1]
```diff
@@ -58,9 +58,11 @@ const Owner = EmberObject.extend(RegistryProxyMixin, ContainerProxyMixin, {
    * @see {@link https://github.com/emberjs/ember.js/blob/v4.5.0-alpha.5/packages/%40ember/engine/instance.ts#L152-L167}
    */
   unregister(fullName: string) {
+    // @ts-expect-error
```
Comment (gitKrystan) [inline addon-test-support/@ember/test-helpers/-internal/build-registry.ts:61]: These types aren't public and there is a bunch of `EmberObject` and `as any` in this file, so I figured leaving `@ts-expect-error` was OK...

File: package.json [hunk 2]
```diff
@@ -60,20 +60,7 @@
...
-    "@types/ember__debug": "~4.0.1",
-    "@types/ember__destroyable": "~4.0.0",
-    "@types/ember__engine": "~4.0.4",
-    "@types/ember__object": "~4.0.2",
-    "@types/ember__polyfills": "~4.0.0",
-    "@types/ember__routing": "~4.0.7",
-    "@types/ember__runloop": "~4.0.1",
-    "@types/ember__service": "~4.0.0",
-    "@types/ember__test": "~4.0.0",
+    "@types/jquery": "^3.5.14",
```
Comment (gitKrystan) [inline package.json:63]: This is needed because `@types/ember-testing-helpers` was removed.

File: tsconfig.json [hunk 3]
```diff
@@ -9,5 +9,5 @@
       "*": ["./types/*"]
     }
   },
-  "include": ["./addon-test-support/**/*.ts", "tests/**/*.ts"]
+  "include": ["addon-test-support/**/*.ts", "tests/**/*.ts", "types/**/*.ts"]
```
Comment (gitKrystan) [inline tsconfig.json:12]: This was necessary for the types to see `types/dummy/index.d.ts`

File: type-tests/tsconfig.json [hunk 4]
```diff
@@ -13,5 +13,6 @@
         "types/*"
       ]
     }
-  }
+  },
+  "include": ["api.ts", "../types/**/*"]
```
Comment (gitKrystan) [inline type-tests/tsconfig.json:17]: This was necessary for the types to see `types/dummy/index.d.ts`

File: addon-test-support/@ember/test-helpers/-internal/build-registry.ts (hunk 1 above)
Comment (chriskrycho) [reply addon-test-support/@ember/test-helpers/-internal/build-registry.ts:61]: That's the best move, yep. We can flag it up for figuring out a non-private-API way of tackling in the future!

**Review by chriskrycho — APPROVED** [review]
Seems good! Thank you!

**Pull request merged**